From 24917516805f372bb414fe5ed15a1cd049291260 Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Fri, 6 Oct 2023 19:49:53 +0100
Subject: [PATCH 73/78] UBUNTU: SAUCE: apparmor4.0.0 [78/82]: apparmor: Make
 apparmor_restrict_unprivileged_unconfined opt-in

BugLink: http://bugs.launchpad.net/bugs/2038567

Existing applications with apparmor confinment have been discovered to
behave incorrectly with the newly added
apparmor_restrict_unprivileged_unconfined feature when enabled by
default with an opt-out. Make this feature opt-in, to unbreak existing
stable production software, most notable LXD.

This security feature can be enabled with:

     sudo sysctl -w kernel.apparmor_restrict_unprivileged_unconfined=1

Signed-off-by: John Johansen <john.johansen@canonical.com>
Signed-off-by: Dimitri John Ledkov <dimitri.ledkov@canonical.com>
Acked-by: Tim Gardner <tim.gardner@canonical.com>
Acked-by: Andrea Righi <andrea.righi@canonical.com>
Signed-off-by: Andrea Righi <andrea.righi@canonical.com>
---
 security/apparmor/policy.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/security/apparmor/policy.c b/security/apparmor/policy.c
index 8eb36124d415..403b596d19f4 100644
--- a/security/apparmor/policy.c
+++ b/security/apparmor/policy.c
@@ -91,7 +91,7 @@ int unprivileged_userns_apparmor_policy = 1;
 int unprivileged_userns_restricted = IS_ENABLED(CONFIG_SECURITY_APPARMOR_RESTRICT_USERNS);
 int unprivileged_userns_restricted_force;
 int unprivileged_userns_restricted_complain;
-int aa_unprivileged_unconfined_restricted = 1;
+int aa_unprivileged_unconfined_restricted = 0;
 int aa_unprivileged_uring_restricted;
 
 const char *const aa_profile_mode_names[] = {
-- 
2.43.0

